lib/rust/flexer/tests/flexer_generated_api_test.rs

```diff
@@ -0,0 +1,524 @@
+    }
+
+    fn gen_state_1_to_3(&mut self) -> StageStatus {
+        match u32::from(self.reader.character()) {
+            97 => StageStatus::ContinueWith(4),
+            98 => StageStatus::ContinueWith(5),
+            _  => {
+                self.current_match = self.reader.pop_result();
+                self.gen_group_1_rule_3();
+                let t = self.matched_bookmark;
+                self.reader.bookmark(t);
+                StageStatus::ExitSuccess
+            }
+        }
+    }
+
+    fn gen_state_1_to_4(&mut self) -> StageStatus {
+        match u32::from(self.reader.character()) {
+            97 => StageStatus::ContinueWith(6),
+            _  => {
+                self.current_match = self.reader.pop_result();
+                self.gen_group_1_rule_0();
+                let t = self.matched_bookmark;
+                self.reader.bookmark(t);
+                StageStatus::ExitSuccess
+            }
+        }
+    }
+
+    fn gen_state_1_to_5(&mut self) -> StageStatus {
+        match u32::from(self.reader.character()) {
+            98 => StageStatus::ContinueWith(7),
+            _  => {
+                self.current_match = self.reader.pop_result();
+                self.gen_group_1_rule_1();
+                let t = self.matched_bookmark;
+                self.reader.bookmark(t);
+                StageStatus::ExitSuccess
+            }
+        }
+    }
+
+    fn gen_state_1_to_6(&mut self) -> StageStatus {
+        match u32::from(self.reader.character()) {
+            97 => StageStatus::ContinueWith(6),
+            _  => {
+                self.current_match = self.reader.pop_result();
+                self.gen_group_1_rule_0();
+                let t = self.matched_bookmark;
+                self.reader.bookmark(t);
+                StageStatus::ExitSuccess
+            }
+        }
+    }
+
+    fn gen_state_1_to_7(&mut self) -> StageStatus {
+        match u32::from(self.reader.character()) {
+            98 => StageStatus::ContinueWith(7),
+            _  => {
+                self.current_match = self.reader.pop_result();
+                self.gen_group_1_rule_1();
+                let t = self.matched_bookmark;
+                self.reader.bookmark(t);
+                StageStatus::ExitSuccess
+            }
+        }
+    }
+
+    fn gen_group_1_rule_0(&mut self) {
+        self.on_spaced_word();
+    }
+
+    fn gen_group_1_rule_1(&mut self) -> () {
+        self.on_spaced_word();
+    }
+
+    fn gen_group_1_rule_2(&mut self) {
+        self.on_no_err_suffix();
+    }
+
+    fn gen_group_1_rule_3(&mut self) {
+        self.on_err_suffix()
+    }
+}
+
+
+// === Trait Impls ===
+
+impl<Reader:LazyReader> Deref for TestLexer<Reader> {
+    type Target = Flexer<TestState,AST,Reader>;
+    fn deref(&self) -> &Self::Target {
+        &self.lexer
+    }
+}
+
+impl<Reader:LazyReader> DerefMut for TestLexer<Reader> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.lexer
+    }
+}
```

Comment:
We put deref impls DIRECTLY below struct definition. Other impls we keep below. Derefs are special because by looking at struct definition you are told that you have access to more fields there, so this information should be next to each other.